File: LeetcodeJava/src/main/java/org/leetcode/algorithm/P3285.java
package org.leetcode.algorithm;

import java.util.ArrayList;
import java.util.List;

/**
 * Author: Deean
 * Date: 2025-03-09 10:21
 * FileName: src/main/java/org/leetcode/algorithm
 * Description: 3285. 找到稳定山的下标
 */

public class P3285 {
    class Solution {
        public List<Integer> stableMountains(int[] height, int threshold) {
            List<Integer> mountains = new ArrayList<>();
            for (int i = 1; i < height.length; i++) {
                if (height[i - 1] > threshold) {
                    mountains.add(i);
                }
            }
            return mountains;
        }
    }

    public static void main(String[] args) {
        Solution s = new P3285().new Solution();
        Object ans = s.stableMountains(new int[]{1, 2, 3, 4, 5}, 2);
        System.out.println(ans);
    }
}

File: LeetcodeCC/algorithm/P3285.cpp
/**
 * Author: Deean
 * Date: 2025-05-12 23:09
 * FileName: P3285.cpp
 * Description: 
 */

#include "../lib/leetcode.h"

using namespace std;

class Solution {
public:
    vector<int> stableMountains(vector<int>& height, int threshold) {
        vector<int> index;
        for (int i = 1; i < height.size(); i++) {
            if (height[i - 1] > threshold) {
                index.push_back(i);
            }
        }
        return index;
    }
};

int main() {
    vector nums = {1, 2, 3, 4, 5};
    vector ans = Solution().stableMountains(nums, 2);
    cout << toString(ans) << endl;
    return 0;
}
